### ISSUE TITLE
Outofmemoryexception in Object Rehash

### ISSUE THREAD

**Issue by mklingen** [CompletelyFairGames/dwarfcorp#465]
https://sentry.io/cfg/dwarfcorp/issues/398489386/

```
OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
  File "C:\Users\matth\Desktop\dwarfcorp\DwarfCorp\DwarfCorpXNA\AssetManagement\GameSave\OverworldFile.cs", line 241, in WriteFile
    Boolean WriteFile(System.String)
  File "C:\Users\matth\Desktop\dwarfcorp\DwarfCorp\DwarfCorpXNA\World\WorldManager.cs", line 570, in SaveThreadRoutine
    Boolean SaveThreadRoutine(System.String)
...
(11 additional frame(s) were not displayed)

Exception of type 'System.OutOfMemoryException' was thrown.
```

**Comment by mklingen** [comment]
This is a difficult one. Possibel the JSON string is just too huge.

**Comment by mklingen** [comment]
Okay, tracked this down to the binary serializer creating intermediate objects. That plus memory fragmentation leads to an out of memory exception Current plan is to replace the overworld datastructure with a set of images, each one containing a layer of data for the overworld. That will make them trivial to save/load.

**Comment by mklingen** [comment]
@Blecki seems to have fixed this.

**Comment by com1clyf3** [comment]
Merged with #484
